supplementary/alica_ros_turtlesim/src/base.cpp

```diff
@@ -13,42 +18,103 @@
 #include <logger/AlicaRosLogger.h>
 #include <ros/ros.h>
 
+#include <boost/dll/import.hpp> // for import_alias
+
 #include <alica_ros_turtlesim/base.hpp>
 
 namespace turtlesim
 {
 
 Base::Base(ros::NodeHandle& nh, ros::NodeHandle& priv_nh, const std::string& name, const int agent_id, const std::string& roleset,
-        const std::string& master_plan, const std::string& path)
+        const std::string& master_plan, const std::string& path, bool loadDynamically)
         : spinner(0)
+        , _loadDynamically(loadDynamically)
 {
-    // create world model
-    ALICATurtleWorldModel::init(nh, priv_nh);
     // Initialize Alica
     ac = new alica::AlicaContext(AlicaContextParams(name, path + "/etc/", roleset, master_plan, false, agent_id));
 
     ac->setCommunicator<alicaRosProxy::AlicaRosCommunication>();
     ac->setTimerFactory<alicaRosTimer::AlicaRosTimerFactory>();
     ac->setLogger<alicaRosLogger::AlicaRosLogger>(agent_id);
+
+    // create world model
+    if (_loadDynamically) {
+        ALICATurtleWorldModelCallInit(nh, priv_nh);
+    } else
+        ALICATurtleWorldModel::init(nh, priv_nh);
+}
+
+void Base::ALICATurtleWorldModelCallInit(ros::NodeHandle& nh, ros::NodeHandle& priv_nh)
```

Comment:
We need to get rid of this singleton for the world model & use `AlicaContext::setWorldModel<>()` instead & the behaviours should access the world model from the engine. I think we need to rework this API to either load the world model dynamically or statically create it as it is done now. 
Can go in a separate PR (need to create a wrike task for it)